src/diagnostics/validateWebSite.ts

```diff
@@ -0,0 +1,81 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { SiteWrapper } from "vscode-azureappservice";
+import TelemetryReporter from "vscode-extension-telemetry";
+import { OutputChannel, ExtensionContext } from 'vscode';
+import { SiteTreeItem } from '../explorer/SiteTreeItem';
+import { IncomingMessage } from 'http';
+import { AzureActionHandler, parseError } from 'vscode-azureextensionui';
+import * as requestP from 'request-promise';
+
+interface IValidateProperties {
+    statusCodes?: string; // Comma-delimited
+    lastStatusCode?: string;
+    lastStatusMessage?: string;
+    timedOut?: 'true' | 'false';
+}
+
+export async function validateWebSite(context: ExtensionContext, siteTreeItem: SiteTreeItem, outputChannel: OutputChannel, telemetryReporter: TelemetryReporter): Promise<void> {
+    let siteWrapper = siteTreeItem.siteWrapper;
+    return (new AzureActionHandler(context, outputChannel, telemetryReporter)).callWithTelemetry('appService.validateWebSite', async (properties: IValidateProperties, _measurements) => {
+        let pollingIntervalMs = 1000;
+        let pollingIncrementMs = 1000; // Increase in interval each time
+        let timeoutTime = Date.now() + 60 * 1000;
+        const uri = siteTreeItem.defaultHostUri;
+        const options: {} = {
+            method: 'GET',
+            uri: uri,
+            resolveWithFullResponse: true
+        };
+        let currentStatusCode: number = 0;
+        let currentStatusMessage: string = '';
+
+        properties.statusCodes = '';
+        properties.timedOut = 'false';
+
+        log(siteWrapper, outputChannel, `Waiting for response from ${uri}.`);
```

Comment:
Prefer to change this to "Validating deployment" or "Verifying site" or something similar. If you just say "waiting for response", I don't know _why_ you're waiting for a response